simphony/cuds/tests/test_cuds.py

```diff
@@ -1,125 +1,223 @@
-
-    def test_add_component_with_no_uuid(self):
-        # Set the uuid to None
-        self.dummpy_component1.uuid = None
-
-        self.cuds.add(self.dummpy_component1)
-
-        self.assertIsNotNone(self.dummpy_component1.uuid)
-        self.assertEqual(self.cuds.get(self.dummpy_component1.uuid),
-                         self.dummpy_component1)
-
-    def test_add_dataset(self):
-        p1 = Particle()
-        p2 = Particle()
+        self.named_cuds_1 = api.Box(name='mybox')
+        self.named_cuds_2 = api.Box(name='mysecondbox')
+        self.nameless_cuds_1 = api.Box()
+
+    def test_cuds_uid(self):
+        c = CUDS()
+
+        self.assertIsNotNone(c.uid)
+        self.assertIsInstance(c.uid, uuid.UUID)
+
+    def test_named_cuds_name(self):
+        c = CUDS(name='mycuds')
+
+        self.assertEqual(c.name, 'mycuds')
+
+    def test_nameless_cuds_name(self):
+        c = CUDS()
+
+        self.assertIsNone(c.name)
+
+    def test_descriptioned_cuds_description(self):
+        c = CUDS(description='test model')
+
+        self.assertEqual(c.description, 'test model')
+
+    def test_descriptionless_cuds_description(self):
+        c = CUDS()
+
+        self.assertIsNone(c.description)
+
+    def test_cuds_data(self):
+        c = CUDS()
+        data = c.data
+
+        self.assertEqual(c.data, data)
+        self.assertIsNot(c.data, data, msg='data is not a copy.')
+        self.assertEqual(len(c.data), 0, msg='data is not empty.')
+
+    def test_add_cuds_component(self):
+        c = CUDS()
+
+        self.assertIsNone(c.add(self.named_cuds_1))
+        self.assertIsNone(c.add(self.nameless_cuds_1))
+
+    def test_add_non_cuds_component(self):
+        c = CUDS()
+
+        self.assertRaises(TypeError, c.add, object())
+
+    def test_add_nameless_cuds_component(self):
+        c = CUDS()
+
+        self.assertIsNone(c.add(self.nameless_cuds_1))
+        self.assertIsNone(c.get(self.nameless_cuds_1.name))
+
+    def test_add_named_cuds_component(self):
+        c = CUDS()
+
+        self.assertIsNone(c.add(self.named_cuds_1))
+        self.assertEqual(c.get(self.named_cuds_1.name),
+                         self.named_cuds_1)
+
+    def test_add_named_component_several_times(self):
+        c = CUDS()
+        c.add(self.named_cuds_1)
+
+        self.assertRaises(ValueError, c.add, self.named_cuds_1)
+
+    def test_add_nameless_component_several_times(self):
+        c = CUDS()
+        c.add(self.nameless_cuds_1)
+        c.add(self.nameless_cuds_1)
+
+        self.assertEqual(c.get_by_uid(self.nameless_cuds_1.uid),
+                         self.nameless_cuds_1)
+        self.assertIsNone(c.get(self.nameless_cuds_1.name))
+
+    def test_get_nameless_cuds_component(self):
+        c = CUDS()
+        c.add(self.nameless_cuds_1)
+
+        self.assertEqual(c.get_by_uid(self.nameless_cuds_1.uid),
+                         self.nameless_cuds_1)
+        self.assertIsNone(c.get(self.nameless_cuds_1.name))
```

Comment:
I guess my comment in CUDS.get is about whether it is a better user-experience to raise an Error here if name is None or empty.